### PR TITLE
[ONNX] Refactor op consistency tests

### LINKED ISSUE #105338: [ONNX] Refactor `test_fx_op_consistency.py`
`test_fx_op_consistency.py` references `ops_test.py` in onnx-script. However, onnx-script tests has refactored to include all dtype tests and prims test. A refactor is needed on `test_fx_op_consistency.py`.
```[tasklist]
### Tasks
- [ ] Use OpInfo rtol/atol
- [ ] Add coverage
- [ ] Prims
```

### PR DESCRIPTION
**Pull request by titaiwangms**
Fixes #105338 

This PR changes the ops consistency tests from manual adding ops into testing list to automated testing all ops in registry. It also spots more complex dtype bugs in the converter.

Overall, this PR provides:
(1) Whole test coverage on ONNX registry
(2) More completed complex supports
(3) Only test the same dtypes as torchlib
(4) Auto xfail unsupported nodes

Follow-up issue: https://github.com/pytorch/pytorch/issues/117118

### CODE REVIEW

File: test/onnx/test_fx_op_consistency.py [hunk 1]
```diff
@@ -629,6 +1414,21 @@ def skip_torchlib_forward_compatibility(
         model_type=pytorch_test_common.TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAM,
         reason="Flaky failure: https://github.com/pytorch/pytorch/issues/115106",
     ),
+    skip(
+        "nn.functional.conv_transpose1d",
+        matcher=lambda sample: isinstance(sample.kwargs.get("padding"), str),
+        reason="String padding is not accepted by aten::conv1d",
```
Comment (justinchuby) [inline test/onnx/test_fx_op_consistency.py:1420]: Curious why this would fail? I thought pytorch will handle the conversion from string to an appropriate attribute

**Review by justinchuby — APPROVED** [review]

**Comment by titaiwangms** [comment]
Hi @huydhn,

This PR makes one of ONNX test exceed 30 minutes restriction, and I see in https://github.com/pytorch/pytorch/blob/71523c22891d769238b71c6720ef976c6cd262dc/test/run_test.py#L606-L617, we could extend it to an hour. Could you point me to the configuration, or how to change the settings? Thanks!

**Review by thiagocrepaldi — COMMENTED** [review]
Maybe try adding a `export TEST_CONFIG=slow-${TEST_CONFIG}` at `.ci/onnx/test.sh` to set the env var?

**Comment by thiagocrepaldi** [comment]
> Maybe try adding a `export TEST_CONFIG=slow-${TEST_CONFIG}` at `.ci/onnx/test.sh` to set the env var?

we may not want to go that route. `slow` pipelines are not run on every commit. Refer to .github/workflows/slow.yml
Maybe there is a way to mark a specific test with a higher timeout instead of marking the pipeline itself as slow

**Comment by huydhn** [comment]
> Hi @huydhn,
> 
> This PR makes one of ONNX test exceed 30 minutes restriction, and I see in
> 
> https://github.com/pytorch/pytorch/blob/71523c22891d769238b71c6720ef976c6cd262dc/test/run_test.py#L606-L617
> , we could extend it to an hour. Could you point me to the configuration, or how to change the settings? Thanks!

Sorry for the late reply.  If a test take more than 5 minutes to finish (IIRC the threshold), it's already way in the realm [slow tests](https://github.com/pytorch/test-infra/blob/generated-stats/stats/slow-tests-by-job.json), and our bot will automatically mark it as such regardless of the timeout value here.

The correct way to handle this IMO is not to increase the global timeout value here, but to decorate that test as slow using this decoration https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_utils.py#L1811-L1823, i.e.
https://github.com/pytorch/pytorch/blob/main/test/torch_np/numpy_tests/linalg/test_linalg.py#L26 and may be create an issue for future investigation.

**Comment by huydhn** [comment]
I'm looking a bit closer into the log and find some really slow tests (>200s), for example:

```
onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_logspace_tensor_overload_cpu_float32 PASSED [244.3099s]`

onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_logspace_tensor_overload_cpu_float16 PASSED [231.7527s] [ 39%]
 onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_logspace_tensor_overload_cpu_complex64 PASSED [217.6158s]

onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_max_pool2d_with_indices_backward_cpu_float16 PASSED [375.7029s]
...
```

They should be marked as slow to avoid timing out.  

@clee2000 Do you know if we run ONNX tests in slow shards?  We could create one if it's not there yet.

File: test/onnx/test_fx_op_consistency.py (hunk 1 above)
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py]: Ah, good catch. The issue here is that something is off when padding="valid" it seems. https://github.com/pytorch/pytorch/issues/117054
